packages/next/src/server/request/params.ts

```diff
@@ -291,7 +298,7 @@ const fallbackParamsProxyHandler: ProxyHandler<Promise<Params>> = {
 
 function makeHangingParams(
   underlyingParams: Params,
-  prerenderStore: PrerenderStoreModern
+  prerenderStore: StaticPrerenderStoreModern
```

Comment:
Let's just make this take a signal so we don't have to invent types like StaticPrerenderStoreModern

Reply:
It's the Exclude that is confusing. I wonder if we ought to just union all the argument site all the PrerenderStoreModerns. I suspect most of them really shouldn't allow client though I haven't checked.

params is a little special b/c we do have special support for "use client" on a layout or page. each layer of unioning and excluding is an abstraction I need to load